File: db4oj/core/src/com/db4o/internal/config/IdSystemConfigurationImpl.java
/* Copyright (C) 2004 - 2010  Versant Inc.  http://www.db4o.com */

package com.db4o.internal.config;

import com.db4o.config.*;
import com.db4o.internal.*;

/**
 * @exclude
 */
public class IdSystemConfigurationImpl implements IdSystemConfiguration {
	
	private final Config4Impl _config;

	public IdSystemConfigurationImpl(Config4Impl config) {
		_config = config;
	}

	public void usePointerBasedSystem() {
		_config.usePointerBasedIdSystem();
	}
	
	public void useStackedBTreeSystem() {
		_config.useStackedBTreeIdSystem();
	}

	public void useInMemorySystem() {
		_config.useInMemoryIdSystem();
		
	}

	public void useCustomSystem(IdSystemFactory factory) {
		_config.useCustomIdSystem(factory);
	}

	public void useSingleBTreeSystem() {
		_config.useSingleBTreeIdSystem();
		
	}

}

File: db4o.net/Db4objects.Db4o/Db4objects.Db4o/Internal/Config/IdSystemConfigurationImpl.cs
/* Copyright (C) 2004 - 2011  Versant Inc.  http://www.db4o.com */

using Db4objects.Db4o.Config;
using Db4objects.Db4o.Internal;

namespace Db4objects.Db4o.Internal.Config
{
	/// <exclude></exclude>
	public class IdSystemConfigurationImpl : IIdSystemConfiguration
	{
		private readonly Config4Impl _config;

		public IdSystemConfigurationImpl(Config4Impl config)
		{
			_config = config;
		}

		public virtual void UsePointerBasedSystem()
		{
			_config.UsePointerBasedIdSystem();
		}

		public virtual void UseStackedBTreeSystem()
		{
			_config.UseStackedBTreeIdSystem();
		}

		public virtual void UseInMemorySystem()
		{
			_config.UseInMemoryIdSystem();
		}

		public virtual void UseCustomSystem(IIdSystemFactory factory)
		{
			_config.UseCustomIdSystem(factory);
		}

		public virtual void UseSingleBTreeSystem()
		{
			_config.UseSingleBTreeIdSystem();
		}
	}
}
